### DISCUSSION TITLE
ItemsRepeater and UniformGridLayout impossible complete alternative AdaptiveGridView

### DISCUSSION THREAD

**Discussion by BigHeadDev** [CommunityToolkit/Windows#495] [Feedback]
ISupportIncrementalLoading is not working properly！
by the way,where is ImageEx in 8.1?

**Comment by Arlodotexe** [comment]
`ImageEx` wasn't ported from 7.x to 8.0, as it was just a thin wrapper with implicit animations and a cache.

`ISupportIncrementalLoading` is an interface and is part of the Platform APIs, what component are you having issues with in the toolkit?

**Reply by BigHeadDev** [reply]
> `ImageEx` wasn't ported from 7.x to 8.0, as it was just a thin wrapper with implicit animations and a cache.
> 
> `ISupportIncrementalLoading` is an interface and is part of the Platform APIs, what component are you having issues with in the toolkit?

Thank you for your reply
1，Will there be support for ImageEx in the future?

2，In the upgrade documentation from 7. x to 8.0, a destructive change is described:
> `AdaptiveGridView` has now been **removed** as it can be **replaced by** the official `ItemRepeater+UniformGridLayout`

But my project cannot be completely replaced by them because I need a **rolling update feature**
The combination of ItemRepeater and UniformGridLayout cannot completely replace AdaptiveGridView
https://learn.microsoft.com/en-us/windows/apps/design/controls/items-repeater#configure-the-data-source
![image](https://github.com/user-attachments/assets/911162ca-ec15-4a9d-aa90-8cbada82689f)
**So, in 8.1, is there any other way to implement AdaptiveGridView with incremental loading support？**

**Reply by Arlodotexe** [reply]
We don't have plans to bring back `AdaptiveGridView` or `ImageEx`. For incremental loading support in ItemsRepeater or UnifromGridLayout, see the `AdvancedCollectionView` in the [Collections](https://github.com/CommunityToolkit/Windows/tree/main/components/Collections/src) component.
